editor/plugins/gradient_editor_plugin.h

```diff
@@ -33,25 +33,106 @@
 
 #include "editor/editor_inspector.h"
 #include "editor/editor_plugin.h"
-#include "gradient_editor.h"
+#include "scene/resources/gradient_texture.h"
 
-class GradientReverseButton : public BaseButton {
-	GDCLASS(GradientReverseButton, BaseButton);
+class EditorSpinSlider;
+class ColorPicker;
+class PopupPanel;
+class GradientTexture1D;
 
-	int margin = 2;
+class GradientEdit : public Control {
+	GDCLASS(GradientEdit, Control);
 
+	Ref<Gradient> gradient;
+	Ref<GradientTexture1D> preview_texture;
+
+	PopupPanel *popup = nullptr;
+	ColorPicker *picker = nullptr;
+
+	bool snap_enabled = false;
+	int snap_count = 10;
+
+	enum GrabMode {
+		GRAB_NONE,
+		GRAB_ADD,
+		GRAB_MOVE
+	};
+
+	GrabMode grabbing = GRAB_NONE;
+	float pre_grab_offset;
+	int pre_grab_index;
+	int selected_index = -1;
+	int hovered_index = -1;
+
+	// Make sure to use the scaled values below.
+	const int BASE_SPACING = 4;
+	const int BASE_HANDLE_WIDTH = 8;
+
+	int draw_spacing = BASE_SPACING;
+	int handle_width = BASE_HANDLE_WIDTH;
+
+	int get_gradient_rect_width();
+
+	void _color_changed(const Color &p_color);
+	void _redraw();
+
+	int get_point_at(int p_xpos);
+	int predict_insertion_index(float p_offset);
+	void show_color_picker();
+
+protected:
+	virtual void gui_input(const Ref<InputEvent> &p_event) override;
 	void _notification(int p_what);
+	static void _bind_methods();
+
+public:
+	void set_gradient(const Ref<Gradient> &p_gradient);
+	Ref<Gradient> get_gradient();
+
+	ColorPicker *get_picker();
+	PopupPanel *get_popup();
+
 	virtual Size2 get_minimum_size() const override;
+
+	void set_selected_index(int p_index);
+
+	void add_point(float p_offset, const Color &p_color);
+	void remove_point(int p_index);
+	void set_offset(int p_index, float p_offset);
+	void set_color(int p_index, const Color &p_color);
+	void reverse_gradient();
+
+	void _reverse_gradient();
```

Comment:
Doesn't look like the public method is implemented, and the underscored method doesn't need to be public.

Reply:
Yeah, it's an artifact of the reverse() method straight up not working because it forgot to sort. Now it can be called outright without any extra steps.